### DISCUSSION TITLE
Unexpected undefined error in Virtualizer._measureElement

### DISCUSSION THREAD

**Discussion by manelio** [TanStack/virtual#576] [General]
After many months using version 3.0.0-beta.54 without errors, after upgrading to 3.0.0-beta.57 I'm getting occasionally this error:

```
utils.mjs:48 Uncaught Error: Unexpected undefined
    at notUndefined (utils.mjs:48:1)
    at Virtualizer._measureElement (index.mjs:402:28)
    at index.mjs:164:1
    at Array.forEach (<anonymous>)
    at ResizeObserver.<anonymous> (index.mjs:163:1)
notUndefined @ utils.mjs:48
Virtualizer._measureElement @ index.mjs:402
(anonymous) @ index.mjs:164
(anonymous) @ index.mjs:163
```

Since traced doesn't reach my code I can't go further.

I'm back to the .54 version for now.

**Comment by HRworksVid** [comment]
Ah, thanks. I tried the other versions in between and getting this error starting from .55. Should we provide an issue or do you have a fix?

**Comment by alanhajo-ui** [comment]
Getting the same issue, even on .54, would be nice with a fix

**Comment by piecyk** [comment]
Please check the latest beta 58

**Reply by HRworksVid** [reply]
Works, thanks!

**Reply by piecyk** [reply]
@HRworksVid can you share an example when it happens?

**Reply by HRworksVid** [reply]
I'll see if I got the time today to provide an MRE. Basically what happened was I got a list of let's say 100 elements. I then scrolled to the bottom and clicked a button which replaced those 100 elements with 50 new ones, so the list was smaller. Scrolling to the bottom or at least >50% is essential for the error to occur.

Context: I am using tanstack/table v8 with this library. My table has lazy loading and [infinite scroll](https://tanstack.com/virtual/v3/docs/examples/react/infinite-scroll). When I load the next batch of table rows (50 -> 100 rows) and scroll down a bit, I then sort a column which replaces my dataset with 50 new rows (sorted from server). Now the error occurs

**Reply by HRworksVid** [reply]
Nevermind, [here you go](https://codesandbox.io/p/sandbox/intelligent-cookies-cdsx7m). Scroll to the bottom and click the button. Keep in mind beta 58 works again, so this example uses beta 57.

@piecyk is this example okay?

**Reply by piecyk** [reply]
@HRworksVid yes, big thanks, the fix is correct 👍
